NasaViewer/ViewControllers/APOD/MainMenuView/APODMainMenuViewController.swift

```diff
@@ -52,23 +53,50 @@ class APODMainMenuViewController: UIViewController {
             contentView.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
         ])
         self.checkforPermission()
+        self.addCancelToButton()
     }
     
     private func pushViewWithRandomArticles() {
-        self.navigationController?.pushViewController(APODTableViewController(randomArticles: 10), animated: true)
+        do {
+            self.navigationController?.pushViewController(try APODTableViewController(randomArticles: 10), animated: true)
+        } catch {
+            self.showError(String(describing: error))
+        }
     }
     
     private func pushViewFromSelectedDay(date: Date) {
-        self.navigationController?.pushViewController(APODTableViewController(articleFromDate: date), animated: true)
+        do {
```

Comment:
you do this do catch dance a lot in those methods. Do you think it's possible to make one generic method for that and call it with desired parameters?